flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/common/CommonExecMatch.java

```diff
@@ -86,7 +89,8 @@ public abstract class CommonExecMatch extends ExecNodeBase<RowData>
 
     public static final String MATCH_TRANSFORMATION = "match";
 
-    private final MatchSpec matchSpec;
+    @JsonProperty(FIELD_NAME_MATCH_SPEC)
```

Comment:
This introduces circular dependency. You use a constant from a child class that you use in the parent class.

Reply:
Could you please also deduplicate the field from `StreamExecMatch` since you're touching this anyhow. I don't think it's necessary to be a separate field in `StreamExecMatch`.